sdk/storage/src/blobs/blob_sas_builder.cpp

```diff
@@ -120,8 +121,9 @@ namespace Azure { namespace Storage { namespace Blobs {
     std::string stringToSign = Permissions + "\n" + (StartsOn.HasValue() ? StartsOn.GetValue() : "")
         + "\n" + ExpiresOn + "\n" + canonicalName + "\n" + Identifier + "\n"
         + (IPRange.HasValue() ? IPRange.GetValue() : "") + "\n" + protocol + "\n" + Version + "\n"
-        + resource + "\n" + Snapshot + "\n" + CacheControl + "\n" + ContentDisposition + "\n"
-        + ContentEncoding + "\n" + ContentLanguage + "\n" + ContentType;
+        + resource + "\n" + (!Snapshot.empty() ? Snapshot : BlobVersionId) + "\n" + CacheControl
```

Comment:
maybe I should choose between snapshot and versionid according to resource type.